### DISCUSSION TITLE
Could someone write a step-by-step guide on how to create failing tests, especially browser ones, in Livewire?

### DISCUSSION THREAD

**Discussion by romansh** [livewire/livewire#6165] [1. Help]
Starting from forking the repository and adding it to Laravel, and so on. I feel stuck and have encountered issues at different stages. It would be easier for me to follow a step-by-step guide rather than trying to explain the specific difficulties I'm facing. I believe it would be helpful for many people in this community.

**Comment by joshhanley** [comment]
@romansh Like this? https://laravel-livewire.com/docs/2.x/contribution-guide

**Reply by joshhanley** [reply]
The only part to skip at the moment is the `npm install` as installing and building the assets is complicated at the moment, due to needing to link to alpine's repo manually (not via npm).

**Reply by romansh** [reply]
I've seen that, but could not understand how to use it. Okay, I have forked it, and what now? How do I connect the forked Livewire folder to the real project where I want to manually test something previously? Or do I not have to do that? And are the forked folder and my project with the bug different pieces of code?

There are a lot of obstacles. Finally, I used Laradock with Selenium and could run regular Dusk tests using the 'regular' folder of my project. So, I could avoid forking but I understand it's the wrong approach.

Even in this case, the main obstacle was how to inherit Test\BrowserTestCase because using \Test\BrowserTestCase did not help, and my class did not see the class.

**Reply by romansh** [reply]
I dropped the idea of working with tests inside Docker and followed the same instructions on the page https://laravel-livewire.com/docs/2.x/contribution-guide, but this time on the host machine (ubuntu 22.04). There were some minor difficulties, but overall, everything went smoothly. I just had to manually install chromedriver since the current dusk-updater is not adapted to work with Chrome 115.

However, currently, many tests are showing the following error:

 
Facebook\WebDriver\Exception\UnknownServerException: unknown error: Chrome failed to start: exited abnormally.
  (unknown error: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /opt/google/chrome/chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
  (Driver info: chromedriver=115.0.5790.170 (cc0d30c2ca5577520c8646671513241faa0bc105-refs/branch-heads/5790@{#1923}),platform=Linux 5.19.0-50-generic x86_64)
After searching the internet, I found some advice on passing parameters to Chrome. A similar approach in Docker for tests via artisan dusk + Selenium worked (I tried this way yesterday for regular Laravel tests).

 
```php 
public function testBasicExample(): void
{
    $options = (new ChromeOptions())->addArguments([
        '--disable-gpu',
        '--headless',
        '--no-sandbox',
        '--window-size=1920,1080',
        '--disable-dev-shm-usage',
        '--disable-extensions',
    ]);

    $driver = RemoteWebDriver::create(
        'http://localhost:9515',
        DesiredCapabilities::chrome()->setCapability(
            ChromeOptions::CAPABILITY,
            $options
        )
    );

    $this->browse(function (Browser $browser) use ($driver) {
        $browser->driver($driver)
            ->visit('/')
            //->waitForLivewire()
            ->assertSee('rely');
    });
}

```

But the error still persists even with those params :(

**Reply by joshhanley** [reply]
@romansh I was recently experiencing issues with the latest chromedriver (115) so had to install 114 instead by running `vendor/bin/dusk-updater update 114 --no-interaction`

**Reply by michael-rubel** [reply]
@romansh I resolved that by installing Chrome `113.0.5672.126` on the local machine, adding the `--headless=new` argument to the Chrome exec, updating the Chromedriver by `./vendor/bin/dusk-updater detect --no-interaction` and starting it manually:
`./vendor/laravel/dusk/bin/chromedriver-linux --verbose --log-path=chromedriver.log &`

**Reply by michael-rubel** [reply]
@joshhanley would you add instructions on how you'd link the Alpine repo?
I have a [PR](https://github.com/livewire/livewire/pull/6831), but the test fails, and it seems it reads the `dist` which we don't have a clue how to compile. 😄
